src/Access/ContextAccess.cpp

```diff
@@ -146,17 +146,22 @@ ContextAccess::ContextAccess(const AccessControl & access_control_, const Params
     : access_control(&access_control_)
     , params(params_)
 {
-    std::lock_guard lock{mutex};
+    setUser(access_control->read<User>(*params.user_id));
```

Comment:
`access_control->setUser()` should be called after calling `access_control->subscribeForChanges()`, otherwise there is a little chance that `user` changes after `setUser()` is returned but before we have a subscription for changes of the `user`. So I suggest to move this call of `setUser()` to `ContextAccess::subscribeForChanges()` and probably it's better to rename `ContextAccess::subscribeForChanges()` to `initialize()`.

Reply:
OK  , I make changes directly on GitHub.